platforms/core-runtime/build-configuration/src/main/java/org/gradle/internal/buildconfiguration/tasks/UpdateDaemonJvmTask.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.buildconfiguration.tasks;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.api.tasks.options.OptionValues;
+import org.gradle.internal.buildconfiguration.UpdateDaemonJvmModifier;
+import org.gradle.internal.jvm.inspection.JvmVendor;
+import org.gradle.internal.jvm.inspection.JvmVendor.KnownJvmVendor;
+import org.gradle.jvm.toolchain.JvmImplementation;
+import org.gradle.work.DisableCachingByDefault;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Generates or updates Daemon JVM criteria.
+ */
+@DisableCachingByDefault(because = "Not worth caching")
+public class UpdateDaemonJvmTask extends DefaultTask {
+
+    public static final String TASK_NAME = "updateDaemonJvm";
```

Comment:
I do not like having the constant here -- in a class that _will_ become public -- because it feels weird for a task class to define a singular task name. 

In practice, Gradle will only instantiate one instance of this task, but technically one could create a number of these tasks with different names. It just seems wrong to me to put a TASK_NAME constant on a type with a `getName` method which could return a different value. 

I think this constant belongs in the plugin that creates the task.